scipy/optimize/tests/test_optimize.py

```diff
@@ -157,7 +157,9 @@ def test_bfgs_nan(self):
         fprime = lambda x: np.ones_like(x)
         x0 = [np.nan]
         with np.errstate(over='ignore', invalid='ignore'):
-            x = optimize.fmin_bfgs(func, x0, fprime, disp=False)
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', LineSearchWarning)
+                x = optimize.fmin_bfgs(func, x0, fprime, disp=False)
```

Comment:
I think the warning should not be filtered at this level, because BFGS actually uses a wrapper to line_search_wolfe2, the function raising the warning. That is `_line_search_wolfe12` in optimize.py, which handles the fact that the line search algorithm may not converge. So, IMO, the warning should be filtered within the wrapper, and it would hence not appear in tests.